### PR TITLE
feat(DENG-2481): Add fenix install referrer to fenix firefox android clients

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(DENG-2481): Add fenix install referrer to fenix firefox android clients

This is part of the effort to enable analysis around DMA related changes.

Changes made:

- Update the query to also pull `play_store_*` attribution info and make it accessible via `firefox_android_clients` view.
- Update the schema.yaml to include the new fields.
- Removed `init.sql`
- Updated the `query.sql` to support / use `is_init()`

### Prior to merging the steps need to be taken:

- Update the schema in BQ

### Post merging we need to:

- Perform a backfill on this table beginning 1st December 2023.

---

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2599)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql [hunk 1]
```diff
@@ -81,7 +89,12 @@ first_session_ping_min_seq AS (
         fenix.first_session AS fenix_first_session
       WHERE
         ping_info.seq IS NOT NULL
-        AND DATE(submission_timestamp) = @submission_date
+        AND
+        {% if is_init() %}
+          DATE(submission_timestamp) >= "2019-06-21"
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:94]: This date doesn't match the others

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql [hunk 2]
```diff
@@ -447,7 +558,59 @@ SELECT
...
+      _current.metadata.install_source__source_ping_datetime
+    ) AS install_source__source_ping_datetime,
+    COALESCE(
+      _previous.metadata.install_source__source_ping_datetime,
+      _current.metadata.install_source__source_ping_datetime
+    ) AS install_source__source_ping_datetime,
+    COALESCE(
+      _previous.metadata.install_source__source_ping_datetime,
+      _current.metadata.install_source__source_ping_datetime
+    ) AS install_source__source_ping_datetime,
```
Comment (fbertsch) [inline sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:589]: copy+paste error?

**Review by fbertsch — APPROVED** [review]
r+wc. Important thing here will be a bit of analysis on the values so it looks like they make sense

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql (hunk 1 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:94]: @fbertsch I'm aware, this appears to be the case in the original `init.sql` so I kept it like that. But I'm also happy to just set all these dates to match.

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:94]: I updated the date to be fixed to `"2019-06-01"`. I hope that works.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql (hunk 2 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql:589]: yes, looks like it 😬 Thanks for catching this.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/schema.yaml [hunk 3]
```diff
@@ -76,6 +76,30 @@ fields:
   mode: NULLABLE
   name: install_source
   type: STRING
+- description: The name of the utm_campaign that is responsible for this installation.
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/schema.yaml:79]: Done.

File: sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql (hunk 1 above)
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql]: Aug. 1, 2020 was when we started the migration to Fenix, so I would consider listing that as the start date: https://docs.telemetry.mozilla.org/concepts/analysis_gotchas.html#notable-historic-events

**Pull request merged**